### PR TITLE
Remove 4.3 deprecated APIs 

### PR DESCRIPTION
**Pull request by youknowriad**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2479,7 +2479,6 @@
 				"@wordpress/compose": "file:packages/compose",
 				"@wordpress/core-data": "file:packages/core-data",
 				"@wordpress/data": "file:packages/data",
-				"@wordpress/deprecated": "file:packages/deprecated",
```
Comment (tofumatt) [inline package-lock.json]: 😱 is that correct?

Comment (tofumatt) [reply package-lock.json]: (I think we added a deprecation in #11543, fwiw.)

**Review by tofumatt — COMMENTED** [review]
I'm curious about removing `@wordpress/deprecated` from the code. Won't we want to continue using it?

File: packages/edit-post/src/store/selectors.js [hunk 2]
```diff
@@ -4,11 +4,6 @@
 import createSelector from 'rememo';
 import { get, includes, some, flatten, values } from 'lodash';
 
-/**
- * WordPress dependencies
- */
-import deprecated from '@wordpress/deprecated';
```
Comment (aduth) [inline packages/edit-post/src/store/selectors.js]: The module is mocked in `test/selectors.js` which should be removed as well.

File: package-lock.json (hunk 1 above)
Comment (aduth) [reply package-lock.json]: It is correct in that the current state of master after this branch would have no occurrence of `@wordpress/deprecated` in `edit-post`.

Comment (youknowriad) [reply package-lock.json]: It's correct as the `edit-post` package don't have any other deprecations anymore.

Ideally, we don't add more deprecations now as we're in API Freeze and keep supporting both APIs unless we think it's very unlikely that someone is using the said API in which I case you can add the dependency back in the PR.

Comment (tofumatt) [reply package-lock.json]: But we could have API changes post-5.0, right? I assumed we'd still want to be able to deprecate APIs after the core merge.

Comment (youknowriad) [reply package-lock.json]: @tofumatt yes, but the process is not defined yet.

**Review by aduth — APPROVED** [review]

File: package-lock.json (hunk 1 above)
Comment (aduth) [reply package-lock.json]: The dependencies should only be those which are actively used. If we need to deprecate in the future, it can be brought back.

**Pull request merged**